clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -834,7 +867,7 @@ class CheckedRegionAdder : public clang::RecursiveASTVisitor<CheckedRegionAdder>
       int Localwild = 0;
 
       for (const auto &SubStmt : S->children()) {
-        CheckedRegionAdder Sub(Context,Writer,Info,Seen);
+        CheckedRegionFinder Sub(Context,Writer,Info,Seen,Map);
```

Comment:
Seems as if you are basically implementing a by-hand bottom-up traversal here. That is, `CheckedRegionFinder` is going to see a statement, but then you'll make another `CheckedRegionFinder` that reaches the child statements. You return `false` from each method to say that the traversal should stop. Is that right?

Reply:
Why make the `visitXXX` methods specific to `for`, or `while`, etc.? I think there's just a general `visitStmt` no?

Reply:
There is, but it seems to override `visitCompoundStmt`

Reply:
Seems to override `visitCompoundStmt` when `visitStmt` is defined

Reply:
OK, then add a `VisitCompoundStatement` then? I.e., aren't `for` loops, `while` loops, etc. all compound statements?